### ISSUE TITLE
Authentication With JWT and Relay

### ISSUE THREAD

**Issue by chris-guidry** [postgraphql/postgraphql#459]
I have followed the documentation regarding [Authentication and Authorization](https://github.com/postgraphql/postgraphql/blob/b8b3f96024ca65288dc8e1adb8cb5e5229979c59/examples/forum/TUTORIAL.md#authentication-and-authorization) to configure my environment. The registration process is working, but I have a few questions regarding authentication.

1. The "authenticate" mutation is returning an object like this: `{"authenticate":{"clientMutationId":"0","jwtToken":{"personId":7,"role":"person"}}}`. I expected jwtToken to be the bearer token. PostGraphQL is being started from node and includes the options `jwtToken: 'public.jwt_token', jwtSecret: 'mySecret'`. What am I missing?
2. In the Relay store the currentPerson field needs to be updated to include the newly authenticated user. Ideally that data would be returned by the authenticate mutation via something like: `      authenticate(input: $input) {
        clientMutationId
        jwtToken {
          personId
          role
        }
        query {
          currentPerson {
            id
            name
          }
        }
      }
`. However, that function requires the JWT Authorization token so it seems like currentPerson would always return null.
3. Once the JWT is actually received, I'm assuming I would update the header that's getting added to the Relay.injectNetworkLayer call?

**Comment by jnsone11** [comment]
1. public.jwt_token must be a PostgreSQL Type declared by you describing your jwt structure, right? Your auth function is returning this jwt_token? Did you include "secret" on postgraphql options? For me jwtSecret didn't work.
2. You can easily decode your jwt and extract your user id. It's just base64 encoded. There are libraries on npm to decode jwt tokens.
3. Yep, right.

**Comment by chris-guidry** [comment]
@jnsone11, thanks for the response. Here is what I found:
1. Yes, public.jwt_token is a PostgreSQL Type as described in the documentation (copied verbatim). I figured out my problem for this item: the parameter for the token is called jwtPgTypeIdentifier, as mentioned in [Using PostGraphQL as a Library](https://github.com/postgraphql/postgraphql/blob/d4fd6a4009fea75dbcaa00d743c985148050475e/docs/library.md#postgraphqlpgconfig-schemaname--public-options). Using jwtSecret works as expected.
2. I was hoping the "authenticate" field could return currentPerson, but it's always null. I'm going to try doing a refetch after I upgrade to Relay Modern.
3. It turns out this part will also require an upgrade to Relay Modern because modern allows for multiple instances of environments so the Relay store will return the correct cache of data depending on the authentication status.

Anyway, thanks for the input!
